algorithms/boyer_moore.go

```diff
@@ -0,0 +1,35 @@
+package algorithms
+
+func Search(needle []rune, haystack []rune) int {
+	n := len(needle)
+	if n == 0 {
+		return -1
+	}
+	maxLen := len(haystack)
+	skipTable := skipTable(needle)
+
+	position := n - 1
+	for position < maxLen {
+		i := n - 1
+		for ; i >= 0 && needle[i] == haystack[position]; i, position = i-1, position-1 {
+		}
+		if i == -1 {
+			return position + 1
+		}
+		if shift, exist := skipTable[haystack[position]]; exist {
```

Comment:
Really sexy solution. This evaluates the existance of a entry in the map and also extracts it.
If the value exists, `exist` ist set to `true` and the value is read into `shift`. If not, `exist` will be `false` and `shift` will be set to `0`